api/client/build.go

```diff
@@ -168,8 +181,12 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	// Dockerfile which uses trusted pulls.
 	context = replaceDockerfileTarWrapper(context, newDockerfile, relDockerfile)
 
+	// clear the progress buffer if -q is provided.
+	if buff, ok := progBuff.(*bytes.Buffer); ok {
+		buff.Reset()
```

Comment:
I'm seeing a difference between when -q is used and when its not.  Try:
```
docker build https://github.com/asdasdasdasd
```
and then try it with `-q` and you'll see that when `-q` is used we don't get the:
```
Downloading build context from remote url: https://github.com/asdasdasdasd 62.73 kB
```
message.  Removing this if-statement shows it.  What was the reason you added this call to Reset() ?